vscode/webviews/UserContextSelector.tsx

```diff
@@ -96,7 +97,12 @@ export const UserContextSelectorComponent: React.FunctionComponent<
                             ? `:${match.range.start.line + 1}-${match.range.end.line + 1}`
                             : ''
                         const description =
-                            match.type === 'file' ? undefined : displayPath(match.uri) + range
+                            match.type === 'file'
+                                ? match.content &&
+                                  estimateTokenCount(match.content) > MAX_CURRENT_FILE_TOKENS
+                                    ? 'Large file will be truncated.'
```

Comment:
```suggestion
                                    ? 'Large file will be truncated.'
```

@toolmantim feel free to update this copy

Reply:
Getting the file content for all search results is most likely going to cause performance issues in large repos / large files, so I'm wondering if we should just show this warning at the top instead.

Reply:
The principle is: don't disappoint/frustrate the user. So we should keep them informed as early as possible. Ideally, tell them what's going on, without bogging down the machine, before they commit to an action.

I suggested stat-ing the filesystem above, which will reduce the load a bit, but stat is not free. So maybe you could just stat items in the viewport and display the labels just in time.

With the way we do context assembly (see detail in #2769) we can't predict accurately whether we'll truncate any given file. So we could inform the user before they pick an item if we're likely going to truncate it, and after the prompt is assembled, too if we did. But we have to do that informing in a way that they can understand—one problem with the first iteration of the error banner is people thought it was a rate limit, etc. We don't want to do that again.

@toolmantim do you have ideas here?

Reply:
updated the behavior based on Tim's figma design